fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/api/LoanTransactionsApiResource.java

```diff
@@ -255,11 +256,34 @@ public String adjustLoanTransaction(@PathParam("loanId") @Parameter(description
             @Parameter(hidden = true) final String apiRequestBodyAsJson) {
 
         final CommandWrapperBuilder builder = new CommandWrapperBuilder().withJson(apiRequestBodyAsJson);
-        final CommandWrapper commandRequest = builder.adjustTransaction(loanId, transactionId).build();
+        CommandProcessingResult result = null;
 
-        final CommandProcessingResult result = this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
+        final CommandWrapper commandRequest = builder.adjustTransaction(loanId, transactionId).build();
+        result = this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
```

Comment:
can you please explain this part?

Reply:
This is adjusting the transaction. Earlier I tried to do undoing part using this API. Will refine this. I've already introduced one for that.